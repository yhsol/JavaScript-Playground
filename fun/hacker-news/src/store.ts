import { NewsFeed, NewsStore } from "./types"

export class Store implements NewsStore {
  private feeds: NewsFeed[];
  private _currentPage: number;

  constructor() {
    this.feeds = [];
    this._currentPage = 1;
  }

  get currentPage() {
    return this._currentPage;
  }

  set currentPage(page: number) {
    this._currentPage = page;
  }

  get nextPage() {
    return this._currentPage + 1;
  }

  get prevPage() {
    return this._currentPage > 1 ? this._currentPage - 1 : 1;
  }
  
  get numberOfFeed() {
    return this.feeds.length;
  }

  get hasFeeds() {
    return this.feeds.length > 0;
  }

  getAllFeeds() {
    return this.feeds;
  }

  getFeed(position: number) {
    return this.feeds[position];
  }

  setFeeds(feeds: NewsFeed[]) {
    this.feeds = feeds.map(feed => ({
      ...feed, 
      read: false,
    }));
  }

  makeRead(id: number) {
    const feed = this.feeds.find(feed => feed.id === id);

    if (feed) {
      feed.read = true;
    }
  }
}