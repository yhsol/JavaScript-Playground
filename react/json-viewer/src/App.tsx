import React from "react";
import JSONNavigatorPage from "./pages/JsonNavigatorPage";

export default function App() {
  return <JSONNavigatorPage />;
}
