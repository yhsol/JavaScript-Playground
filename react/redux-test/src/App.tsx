import { Route, Routes, Link } from "react-router-dom";
import Main from "./pages/Main";
import Layout from "./components/layout/Layout";
import About from "./pages/About";
import Article from "./pages/Article";
import Articles from "./pages/Articles";
import Login from "./pages/Login";
import MyPage from "./pages/MyPage";
import NotFound from "./pages/NotFound";
import Profile from "./pages/Profile";
import Todo from "./pages/Todo";
import Stat from "./pages/Stat";
import CharacterCounter from "./pages/CharacterCounter";
import RecoilTodo from "./pages/RecoilTodo";
// import { lazy } from "react";
// const About = lazy(() => import("./pages/About"))
// const Article = lazy(() => import("./pages/Article"))
// const Articles = lazy(() => import("./pages/Articles"))
// const Login = lazy(() => import("./pages/Login"))
// const MyPage = lazy(() => import("./pages/MyPage"))
// const NotFound = lazy(() => import("./pages/NotFound"))
// const Profile = lazy(() => import("./pages/Profile"))
// const Todo = lazy(() => import("./pages/Todo"))

function App() {
  return (
    <>
      <div>
        <Link to={"/"}>Main</Link>
        <Link to={"/about"}>About</Link>
        <Link to={"/todo"}>Todo</Link>
      </div>
      <Routes>
        <Route element={<Layout />}>
          <Route path={"/"} element={<Main />} />
          <Route path={"/about"} element={<About />}></Route>
          <Route path={"/profiles/:username"} element={<Profile />}></Route>
          <Route path={"/todo"} element={<Todo />}></Route>
          <Route path={"/stat"} element={<Stat />}></Route>
          <Route path={"/character"} element={<CharacterCounter />}></Route>
          <Route path={"/recoil-todo"} element={<RecoilTodo />}></Route>
        </Route>
        <Route path={"/articles"} element={<Articles />}>
          <Route path={":id"} element={<Article />}></Route>
        </Route>
        <Route path={"/login"} element={<Login />}></Route>
        <Route path={"/mypage"} element={<MyPage />}></Route>
        <Route path="*" element={<NotFound />}></Route>
      </Routes>
    </>
  );
}

export default App;
