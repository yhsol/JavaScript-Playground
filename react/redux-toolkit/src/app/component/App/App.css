.App {
  padding: 2rem;
  display: flex;
  flex-wrap: wrap;
  margin-right: -2rem;
}

.App > * {
  width: 50rem;
  margin-right: 2rem;
  margin-bottom: 2rem;
}

h5 {
  margin-bottom: 0 !important;
}
