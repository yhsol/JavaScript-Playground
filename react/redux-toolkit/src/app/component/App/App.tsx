import React from "react";
import ProfileCard from "../../../features/profile/components/ProfileCard/ProfileCard";
import TodoCard from "../../../features/todo/components/TodoCard/TodoCard";

const App = () => {
  return (
    <div className="App">
      <ProfileCard />
      <TodoCard />
    </div>
  );
};

export default App;
