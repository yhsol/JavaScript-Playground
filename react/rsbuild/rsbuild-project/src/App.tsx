import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { List } from "./virtualized-scrolling/List";
import { useDictionary } from "./virtualized-scrolling/useDictionary";

const queryClient = new QueryClient();

const App = () => {
  const items = useDictionary();
  return (
    <QueryClientProvider client={queryClient}>
      {/* <StepFormProvider>
        <StepForm />
        <CheckStepForm />
      </StepFormProvider> */}
      {/* <ReusableModalProvider>
        <ReusableModal />
        <CompUsingReusableModal />
      </ReusableModalProvider> */}
      <List items={items} />
    </QueryClientProvider>
  );
};

export default App;
