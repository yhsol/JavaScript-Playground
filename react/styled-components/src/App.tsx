import React from "react";
import GridList from "./components/GridList";
import List from "./components/List";

function App() {
  return (
    <div>
      <GridList />
      <List />
    </div>
  );
}

export default App;
