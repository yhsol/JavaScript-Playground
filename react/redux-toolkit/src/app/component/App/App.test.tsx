import {
  fireEvent,
  render,
  RenderResult,
  screen,
  within,
} from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../store";
import App from "./App";

const renderApp = (): RenderResult =>
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );

test("renders the app", () => {
  const { asFragment } = renderApp();
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByLabelText(/profile card/i)).toBeInTheDocument();
  expect(screen.getByLabelText(/todo card/i)).toBeInTheDocument();
});

test("handles profile click", () => {
  const { asFragment } = renderApp();
  const profileCard = screen.getByLabelText(/profile card/i);
  const profileList = within(profileCard).getByRole("list");
  const profileListItems = within(profileList).getAllByRole("listitem");

  const todoCard = screen.getByLabelText(/todo card/i);
  const todoList = within(todoCard).getByRole("list");
  let todoListItems = within(todoList).queryAllByRole("lsititem");

  fireEvent.click(profileListItems[0]);
  todoListItems = within(todoList).queryAllByRole("listitem");
  expect(todoListItems).toHaveLength(3);
  expect(asFragment()).toMatchSnapshot();
});

test("handles profile delete click", () => {
  const { asFragment } = renderApp();
  const profileCard = screen.getByLabelText(/profile card/i);
  const profileList = within(profileCard).getByRole("list");
  const profileListItems = within(profileList).getAllByRole("listitem");
  fireEvent.click(profileListItems[0]);

  const profileDeleteButton = within(profileListItems[0]).getByText(/delete/i);

  fireEvent.click(profileDeleteButton);
  const todoCard = screen.getByLabelText(/todo card/i);
  const todoList = within(todoCard).getByRole("list");
  const todoListItems = within(todoList).queryAllByRole("listitem");
  expect(todoListItems).toHaveLength(0);
  expect(asFragment()).toMatchSnapshot();
});

test("handles create new todo click", () => {
  const { rerender } = renderApp();
  let alert = screen.queryByRole("alert");
  expect(alert).not.toBeInTheDocument();
  const createNewTodoButton = screen.getByText(/create new todo/i);
  fireEvent.click(createNewTodoButton);
  rerender(
    <Provider store={store}>
      <App />
    </Provider>
  );
  alert = screen.getByRole("alert");
  expect(alert).toBeInTheDocument();
  const alertCloseButton = within(alert).getByRole("button");
  fireEvent.click(alertCloseButton);
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(alert).not.toBeInTheDocument();
});
