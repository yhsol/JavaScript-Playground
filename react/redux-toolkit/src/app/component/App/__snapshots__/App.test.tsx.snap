// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles profile click 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      aria-label="Profile Card"
      class="card"
      role="main"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Profiles
        </h5>
      </div>
      <div
        class="card-body"
      >
        <ul
          class="list-group"
        >
          <li
            class="list-group-item profileItem active"
          >
            <div>
              <label
                for="name-1-input"
              >
                Name
              </label>
              <input
                autofocus=""
                class="form-control"
                id="name-1-input"
                type="text"
                value="Ben"
              />
            </div>
            <div>
              <label
                for="type-1-select"
              >
                Type
              </label>
              <select
                class="form-control"
                id="type-1-select"
              >
                <option>
                  guest
                </option>
                <option>
                  user
                </option>
                <option>
                  admin
                </option>
              </select>
            </div>
            <div
              class="buttonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
          <li
            class="list-group-item profileItem"
          >
            <div>
              <label
                for="name-2-input"
              >
                Name
              </label>
              <input
                class="form-control"
                id="name-2-input"
                type="text"
                value="Sue"
              />
            </div>
            <div>
              <label
                for="type-2-select"
              >
                Type
              </label>
              <select
                class="form-control"
                id="type-2-select"
              >
                <option>
                  guest
                </option>
                <option>
                  user
                </option>
                <option>
                  admin
                </option>
              </select>
            </div>
            <div
              class="buttonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="card-footer"
      >
        <div
          class="wrapper"
        >
          <button
            class="btn btn-primary"
          >
            Create New Profile
          </button>
        </div>
      </div>
    </div>
    <div
      aria-label="Todo Card"
      class="card"
      role="complementary"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Todos for Ben
        </h5>
      </div>
      <div
        class="card-body"
      >
        <ul
          class="list-group"
        >
          <li
            class="list-group-item todoItem"
          >
            <div>
              <label
                for="isComplete-1-input"
              >
                Complete?
              </label>
              <input
                checked=""
                class="form-control checkbox"
                id="isComplete-1-input"
                type="checkbox"
              />
            </div>
            <div>
              <label
                for="description-1-input"
              >
                Description
              </label>
              <input
                class="form-control"
                id="description-1-input"
                type="text"
                value="eat tacos"
              />
            </div>
            <div
              class="butonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
          <li
            class="list-group-item todoItem"
          >
            <div>
              <label
                for="isComplete-2-input"
              >
                Complete?
              </label>
              <input
                class="form-control checkbox"
                id="isComplete-2-input"
                type="checkbox"
              />
            </div>
            <div>
              <label
                for="description-2-input"
              >
                Description
              </label>
              <input
                class="form-control"
                id="description-2-input"
                type="text"
                value="drink milk"
              />
            </div>
            <div
              class="butonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
          <li
            class="list-group-item todoItem"
          >
            <div>
              <label
                for="isComplete-3-input"
              >
                Complete?
              </label>
              <input
                class="form-control checkbox"
                id="isComplete-3-input"
                type="checkbox"
              />
            </div>
            <div>
              <label
                for="description-3-input"
              >
                Description
              </label>
              <input
                class="form-control"
                id="description-3-input"
                type="text"
                value="walk and chew gum"
              />
            </div>
            <div
              class="butonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="card-footer"
      >
        <div
          class="wrapper"
        >
          <button
            class="btn btn-primary"
          >
            Create New Todo
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`handles profile delete click 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      aria-label="Profile Card"
      class="card"
      role="main"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Profiles
        </h5>
      </div>
      <div
        class="card-body"
      >
        <ul
          class="list-group"
        >
          <li
            class="list-group-item profileItem"
          >
            <div>
              <label
                for="name-2-input"
              >
                Name
              </label>
              <input
                class="form-control"
                id="name-2-input"
                type="text"
                value="Sue"
              />
            </div>
            <div>
              <label
                for="type-2-select"
              >
                Type
              </label>
              <select
                class="form-control"
                id="type-2-select"
              >
                <option>
                  guest
                </option>
                <option>
                  user
                </option>
                <option>
                  admin
                </option>
              </select>
            </div>
            <div
              class="buttonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="card-footer"
      >
        <div
          class="wrapper"
        >
          <button
            class="btn btn-primary"
          >
            Create New Profile
          </button>
        </div>
      </div>
    </div>
    <div
      aria-label="Todo Card"
      class="card"
      role="complementary"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Todos 
        </h5>
      </div>
      <div
        class="card-body"
      >
        <ul
          class="list-group"
        />
      </div>
      <div
        class="card-footer"
      >
        <div
          class="wrapper"
        >
          <button
            class="btn btn-primary disabled"
          >
            Create New Todo
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders the app 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      aria-label="Profile Card"
      class="card"
      role="main"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Profiles
        </h5>
      </div>
      <div
        class="card-body"
      >
        <ul
          class="list-group"
        >
          <li
            class="list-group-item profileItem"
          >
            <div>
              <label
                for="name-1-input"
              >
                Name
              </label>
              <input
                class="form-control"
                id="name-1-input"
                type="text"
                value="Ben"
              />
            </div>
            <div>
              <label
                for="type-1-select"
              >
                Type
              </label>
              <select
                class="form-control"
                id="type-1-select"
              >
                <option>
                  guest
                </option>
                <option>
                  user
                </option>
                <option>
                  admin
                </option>
              </select>
            </div>
            <div
              class="buttonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
          <li
            class="list-group-item profileItem"
          >
            <div>
              <label
                for="name-2-input"
              >
                Name
              </label>
              <input
                class="form-control"
                id="name-2-input"
                type="text"
                value="Sue"
              />
            </div>
            <div>
              <label
                for="type-2-select"
              >
                Type
              </label>
              <select
                class="form-control"
                id="type-2-select"
              >
                <option>
                  guest
                </option>
                <option>
                  user
                </option>
                <option>
                  admin
                </option>
              </select>
            </div>
            <div
              class="buttonWrapper"
            >
              <button
                class="btn btn-danger"
              >
                Delete
              </button>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="card-footer"
      >
        <div
          class="wrapper"
        >
          <button
            class="btn btn-primary"
          >
            Create New Profile
          </button>
        </div>
      </div>
    </div>
    <div
      aria-label="Todo Card"
      class="card"
      role="complementary"
    >
      <div
        class="card-header"
      >
        <h5
          class="card-title"
        >
          Todos 
        </h5>
      </div>
      <div
        class="card-body"
      >
        <ul
          class="list-group"
        />
      </div>
      <div
        class="card-footer"
      >
        <div
          class="wrapper"
        >
          <button
            class="btn btn-primary disabled"
          >
            Create New Todo
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
